react/src/components/timeline-app/TimelineComponent.jsx

```diff
@@ -0,0 +1,177 @@
+      if (event.description) {
+        event.description = '<p>' + event.description.replace(/\n/g, '</p><p>') + '</p>';
+      }
+      return {
+        max: localMax > max ? localMax : max,
+        min: event.start < min ? event.start : min,
+      };
+    }, { max: -Infinity, min: Infinity });
+    const margin = max.getYear() - min.getYear();
+
+    /* Display the vis-timeline */
+    const container = this.refs.visTimeline;
+    const items = new vis.DataSet(events);
+    const options = {
+      align: 'left',
+      format: {
+        minorLabels: {
+          millisecond: '',
+          second: '',
+          minute: '',
+          hour: '',
+          weekday: 'ddd D',
+          day: 'D',
+          month: 'MMM',
+          year: 'YYYY',
+        },
+        majorLabels: {
+          millisecond: 'ddd D MMMM YYYY',
+          second: 'ddd D MMMM YYYY',
+          minute: 'ddd D MMMM YYYY',
+          hour: 'ddd D MMMM YYYY',
+          weekday: 'MMMM YYYY',
+          day: 'MMMM YYYY',
+          month: 'YYYY',
+          year: '',
+        },
+      },
+      maxHeight: '25em',
+      minHeight: '10em',
+      orientation: 'top',
+      selectable: false,
+      showCurrentTime: false,
+      template: (event, element) => {
+        const className = 'timeline__itembox ' +
+          (event.icon ? 'timeline__itembox--hasicon ' : '') +
+          (event.linkType === 'external' ? 'timeline__itembox--external ' : '');
+        const html = (
+          <div>
+            <div className={ className } id={ 'timeline__itembox__' + event.id }>
+              { event.icon ? <div className="timeline__itembox__imagebox"><img src={ event.icon } /></div> : '' }
+              <div className="timeline__itembox__titlebox">
+                {
+                  event.link ?
+                    <a href={ event.link } target={ event.linkType === 'external' ? '_blank' : '_self' }>
+                      { event.title }
+                    </a>
+                    : event.title
+                }
+              </div>
+            </div>
+          </div>
+        );
+        return ReactDOM.render(html, element);
+      },
+      zoomMin: 1000 * 60 * 60 * 24 * 3,
+      zoomMax: 1000 * 60 * 60 * 24 * 365 * (margin + 10) * 2,
+    };
+    const timeline = new vis.Timeline(container, items, options);
+
+    /* Create the tooltips */
+    events.forEach((event) => {
+      const date = event.start;
+      let tooltipContent = `<p class="timeline__tooltip__date">${date.getDate()}.${date.getMonth() + 1}.${date.getFullYear()}</p>`;
+      if (event.description) {
+        tooltipContent += event.description;
+      }
+      tooltipContent += `<span class="timeline__tooltip__arrow"></span>`;
+      const tooltipNode = document.createElement('div');
+      tooltipNode.innerHTML = tooltipContent;
+      tooltipNode.classList.add('timeline__tooltip');
+      tooltipNode.id = 'timeline__tooltip__' + event.id;
+      container.appendChild(tooltipNode);
```

Comment:
I think we could abstract this in a React component? Is that possible? And use in the render() function?
```jsx
render() { 
    return ( 
       <div>
          {events.map(event => <ToolTip event={event} />}
      </div>
  )
}
```